pkg/comp-functions/functions/vshnminio/minio_deploy.go

```diff
@@ -0,0 +1,226 @@
+				"cpu":    cpu,
+			},
+		},
+		"persistence": map[string]interface{}{
+			"size": disk,
+		},
+	}
+
+	vb, err := json.Marshal(values)
+	if err != nil {
+		return err
+	}
+
+	r := &xhelmbeta1.Release{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: comp.GetName(),
+		},
+		Spec: xhelmbeta1.ReleaseSpec{
+			ForProvider: xhelmbeta1.ReleaseParameters{
+				Chart: xhelmbeta1.ChartSpec{
+					Repository: iof.Config.Data["minioChartRepository"],
+					Version:    iof.Config.Data["minioChartVersion"],
+					Name:       "minio",
+				},
+				Namespace: comp.GetInstanceNamespace(),
+				ValuesSpec: xhelmbeta1.ValuesSpec{
+					Values: k8sruntime.RawExtension{
+						Raw: vb,
+					},
+				},
+			},
+			ResourceSpec: xpv1.ResourceSpec{
+				ProviderConfigReference: &xpv1.Reference{
+					Name: "helm",
+				},
+				WriteConnectionSecretToReference: &xpv1.SecretReference{
+					Name:      comp.GetName() + "-connection",
+					Namespace: comp.GetInstanceNamespace(),
+				},
+			},
+			ConnectionDetails: []xhelmbeta1.ConnectionDetail{
+				{
+					ObjectReference: corev1.ObjectReference{
+						APIVersion: "v1",
+						Kind:       "Secret",
+						Name:       comp.GetName() + "-minio",
+						Namespace:  comp.GetInstanceNamespace(),
+						FieldPath:  "data.rootUser",
+					},
+					ToConnectionSecretKey: "MINIO_USERNAME",
+				},
+				{
+					ObjectReference: corev1.ObjectReference{
+						APIVersion: "v1",
+						Kind:       "Secret",
+						Name:       comp.GetName() + "-minio",
+						Namespace:  comp.GetInstanceNamespace(),
+						FieldPath:  "data.rootPassword",
+					},
+					ToConnectionSecretKey: "MINIO_PASSWORD",
+				},
+			},
+		},
+	}
+
+	cd := []v1alpha1.DerivedConnectionDetail{
+		{
+			Name:                    pointer.String("MINIO_USERNAME"),
+			FromConnectionSecretKey: pointer.String("MINIO_USERNAME"),
+			Type:                    v1alpha1.ConnectionDetailTypeFromConnectionSecretKey,
+		},
+		{
+			Name:                    pointer.String("MINIO_PASSWORD"),
+			FromConnectionSecretKey: pointer.String("MINIO_PASSWORD"),
+			Type:                    v1alpha1.ConnectionDetailTypeFromConnectionSecretKey,
+		},
+	}
+
+	return iof.Desired.PutWithResourceName(ctx, r, comp.Name+"-release", runtime.AddDerivedConnectionDetails(cd))
+}
+
+func createServiceObserver(ctx context.Context, comp *vshnv1.VSHNMinio, iof *runtime.Runtime) error {
+
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName() + "-minio",
+			Namespace: comp.GetInstanceNamespace(),
+		},
+	}
+
+	return iof.Desired.PutIntoObserveOnlyObject(ctx, service, comp.Name+"-service-observer")
+}
+
+func getConnectionDetails(ctx context.Context, comp *vshnv1.VSHNMinio, iof *runtime.Runtime) error {
+
+	service := &corev1.Service{}
+
+	err := iof.Observed.GetFromObject(ctx, service, comp.Name+"-service-observer")
+	if err != nil {
+		return err
+	}
+
+	iof.Desired.PutCompositeConnectionDetail(ctx, v1alpha1.ExplicitConnectionDetail{
+		Name:  "MINIO_HOST",
```

Comment:
I think this should be URL and not the HOST. We have to make sure we are consistent with [that](https://docs.appcat.ch/vshn-managed/postgresql/create.html#_find_the_connection_details).

Reply:
Fixed